### DISCUSSION TITLE
Question about importing a Youtube video into OTIO

### DISCUSSION THREAD

**Discussion by utsab** [AcademySoftwareFoundation/OpenTimelineIO#981] [Q&A]
I and @hisergiorojas are working on creating an example tutorial for OTIO which will overlay some useful data onto the timeline for a youtube video.  (I presented about this idea at the last OTIO group meeting).  

We have successfully used youtube-dl to download the video (as an .mp4 file) as well as the Youtube description.  We are not sure what to do next in order to import this data into OTIO.  

It seems that we need to somehow create a timeline file (as an .xml or .edl file for example) and then feed this into OTIO.  Is this correct?  How would we create such a file?  

Thank you.

**Comment by KarthikRIyer** [comment]
I think you could parse the video description with python and use the otio python api to create a timeline object. IIRC you're trying to recreate the chapters feature from youtube. Maybe you could use `Marker` to store the chapters.

Then use the adapter system to write an otio file if necessary:
`otio.adapters.write_to_file(mytimeline, "something.otio")`

Is that what you're looking for?

**Reply by jminor** [reply]
Yes, this is a good approach.

Here is some Python example code which creates a Timeline and some Clips from a single movie file which might be useful for reference: https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/examples/shot_detect.py

In your case, you could choose to make a Clip for each chapter, or just one long clip for the whole thing but with Markers for each chapter, as Karthik suggested. Either one seems reasonable.

**Reply by utsab** [reply]
Thank you @KarthikRIyer and @jminor.  Yes, representing the youtube chapters as OTIO markers makes sense.

To clarify the end-to-end workflow for the tutorial, here is what we are imagining:
1. The starting input would be an existing video on youtube.
2. We download the video and youtube description using youtube-dl
3. Parse the video description in python and create a Timeline object with Markers to represent the chapters
4. Create a single clip on the timeline with a media_reference to the original youtube video file
5. Export the timeline as a .otio file

So in other words, we start with the original youtube video and description, and the end result would be an otio file that represents this information as a Timeline.

What do you think?  Is this a reasonable approach?
